### PR TITLE
Update Main Branch Changes with Azure Storage Track2 Methods

### PR DESCRIPTION
**Pull request by nytian**
Branch `nytian/updateasv12`: Syncs updates between the main branch and azure-storage-v12.
Branch `nytian/merge_main` (feature branch): Built on branch `nytian/updateasv12`, this PR includes adjustments necessary for the integration of previous updates into Azure Storage Track2.

After this PR is approved and merged, then I will merge branch `nytian/updateasv12` to `azure-storage-v12`.

### CODE REVIEW

File: src/DurableTask.AzureStorage/Tracking/OrchestrationInstanceStatusQueryCondition.cs [hunk 1]
```diff
@@ -145,10 +145,7 @@ internal ODataCondition ToOData()
             }
             else if (this.ExcludeEntities)
             {
-                conditions.Add(TableQuery.CombineFilters(
-                     TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.LessThan, "@"),
-                     TableOperators.Or,
-                     TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.GreaterThanOrEqual, "A")));
+                conditions.Add($"{nameof(OrchestrationInstanceStatus.PartitionKey)} lt '@' or {nameof(OrchestrationInstanceStatus.PartitionKey)} ge 'A'");
```
Comment (nytian) [inline src/DurableTask.AzureStorage/Tracking/OrchestrationInstanceStatusQueryCondition.cs:148]: In Azure Storage Track2, we don't have TableQuery anymore, so I translate the original query with this to align with other 
 context.

File: src/DurableTask.AzureStorage/EntityTrackingStoreQueries.cs [hunk 2]
```diff
@@ -56,7 +57,7 @@ class EntityTrackingStoreQueries : EntityBackendQueries
             CancellationToken cancellation = default(CancellationToken))
         {
             await this.ensureTaskHub();
-            OrchestrationState? state = (await this.trackingStore.GetStateAsync(id.ToString(), allExecutions: false, fetchInput: includeState)).FirstOrDefault();
+            OrchestrationState? state = await this.trackingStore.GetStateAsync(id.ToString(), allExecutions: false, fetchInput: includeState).FirstOrDefaultAsync();
```
Comment (nytian) [inline src/DurableTask.AzureStorage/EntityTrackingStoreQueries.cs:60]: I can only pass the build by updating FirstOrDefault to Async

File: src/DurableTask.AzureStorage/EntityTrackingStoreQueries.cs [hunk 3]
```diff
@@ -87,7 +88,10 @@ public async override Task<EntityQueryResult> QueryEntitiesAsync(EntityQuery fil
 
             do
             {
-                DurableStatusQueryResult result = await this.trackingStore.GetStateAsync(condition, filter.PageSize ?? 100, continuationToken, cancellation);
+                Page<OrchestrationState>? page = await this.trackingStore.GetStateAsync(condition, cancellation).AsPages(continuationToken, filter.PageSize ?? 100).FirstOrDefaultAsync();
+                DurableStatusQueryResult result = page != null
+                    ? new DurableStatusQueryResult { ContinuationToken = page.ContinuationToken, OrchestrationState = page.Values }
+                    : new DurableStatusQueryResult { OrchestrationState = Array.Empty<OrchestrationState>() };
```
Comment (nytian) [inline src/DurableTask.AzureStorage/EntityTrackingStoreQueries.cs:94]: In AS track2, there is no GetStateAsync definition that can return `OrchestrationState`, so I use [this definition](https://github.com/Azure/durabletask/blame/azure-storage-v12/src/DurableTask.AzureStorage/Tracking/AzureTableTrackingStore.cs#L531) to convert a OrchestrationState result.

File: src/DurableTask.AzureStorage/Tracking/OrchestrationInstanceStatusQueryCondition.cs (hunk 1 above)
Comment (davidmrdavid) [reply src/DurableTask.AzureStorage/Tracking/OrchestrationInstanceStatusQueryCondition.cs:148]: can you please add a comment on top of this method that documents the query? It's a big hard to follow as-is

File: src/DurableTask.AzureStorage/EntityTrackingStoreQueries.cs (hunk 3 above)
Comment (davidmrdavid) [reply src/DurableTask.AzureStorage/EntityTrackingStoreQueries.cs:94]: Do we have other examples of this new `GetStateAsync` method in the codebase? Just trying to confirm this is a pattern already in the branch

**Review by davidmrdavid — COMMENTED** [review]
Left a few comments, but mostly looks good!

File: src/DurableTask.AzureStorage/EntityTrackingStoreQueries.cs (hunk 3 above)
Comment (nytian) [reply src/DurableTask.AzureStorage/EntityTrackingStoreQueries.cs:94]: A similar case is here from azure-storage-v12: https://github.com/Azure/durabletask/blob/azure-storage-v12/src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs#L1790

File: src/DurableTask.AzureStorage/Tracking/OrchestrationInstanceStatusQueryCondition.cs (hunk 1 above)
Comment (nytian) [reply src/DurableTask.AzureStorage/Tracking/OrchestrationInstanceStatusQueryCondition.cs:148]: Just want to confirm you mean what `$"{nameof(OrchestrationInstanceStatus.PartitionKey)} lt '@' or {nameof(OrchestrationInstanceStatus.PartitionKey)} ge 'A'")` means, right?

**Review by davidmrdavid — APPROVED** [review]
